src/main/java/eu/enexa/docker/ContainerManagerImpl.java

```diff
@@ -66,25 +73,35 @@ public String startContainer(String image, String containerName, List<AbstractMa
 
             List<Volume> volumes = new ArrayList<>();
             String expIRI="";
+            String ContainerWritablePath ="";
+            String ContainerModuleInstancePath ="";
             for(AbstractMap.SimpleEntry v:variables){
                 if(v.getKey().toString().equals("ENEXA_EXPERIMENT_IRI")){
                     expIRI = v.getValue().toString();
                 }
+                if(v.getKey().toString().equals("ENEXA_WRITEABLE_DIRECTORY")){
+                    ContainerWritablePath = v.getValue().toString();
+                }
+                if(v.getKey().toString().equals("ENEXA_MODULE_INSTANCE_DIRECTORY")){
+                    ContainerModuleInstancePath = v.getValue().toString();
+                }
             }
             if(expIRI.equals("")||expIRI.length()<10){
                 LOGGER.warn("ENEXA_EXPERIMENT_IRI is null or less than 10 character");
             }
 
-            //String HOST_PATH = HOST_Base_PATH.concat(expIRI.replace("http://",""));
-            String HOST_PATH = HOST_BASE_PATH;
-            /*if(!HOST_Base_PATH.endsWith(File.separator)){
-                HOST_PATH = HOST_Base_PATH.concat(File.separator.concat(expIRI.replace("http://","")));
-            }*/
-
             List<Bind> allBinds = new ArrayList<>();
             //allBinds.add(new Bind(HOST_PATH, new Volume(VOLUME_PATH+"/"+expIRI.replace("http://",""))));
             //allBinds.add(new Bind(HOST_PATH+"/output", new Volume("/output")));
-            allBinds.add(new Bind(HOST_PATH, new Volume("/home/shared")));
+            allBinds.add(new Bind(hostBasePath, new Volume("/home/shared"),AccessMode.ro));
+            //allBinds.add(new Bind(hostWritablePath, new Volume("/home/writeable"),AccessMode.rw));
+            allBinds.add(new Bind(hostWritablePath, new Volume(ContainerWritablePath),AccessMode.rw));
+            //allBinds.add(new Bind(hostModuleInstancePath, new Volume("/tmp"),AccessMode.rw));
+            allBinds.add(new Bind(hostModuleInstancePath, new Volume(ContainerModuleInstancePath),AccessMode.rw));
```

Comment:
This third binding is not necessary since the module instance path is a sub path of the writable path.